scanner/keppel/scanner/scanner.go

```diff
@@ -221,6 +236,36 @@ func (s *Scanner) GetTrivyReport(account string, repository string, manifest str
 
 }
 
+// extractImageInfo extracts image registry, image repository and the account name
+// from a container image
+func (s *Scanner) ExtractImageInfo(image string) (ImageInfo, error) {
+	// Split the string to remove the tag
+	parts := strings.Split(image, ":")
+	if len(parts) < 1 {
+		return ImageInfo{}, fmt.Errorf("invalid image")
+	}
+
+	// Split the remaining string by '/'
+	components := strings.Split(parts[0], "/")
+	if len(components) < 3 || len(components) > 4 {
+		return ImageInfo{}, fmt.Errorf("invalid image string format: expected 3 or 4 components")
+	}
+
+	info := ImageInfo{
+		Registry:   components[0],
+		Repository: components[len(components)-1],
+	}
+
+	if len(components) == 3 {
+		info.Account = components[1]
+	} else { // len(components) == 4
+		info.Account = components[1]
+		info.Organization = components[2]
```

Comment:
Hmm why do we keep that "organization" separate? 

Actually, that's artificial due to the mirroring. I would maybe rather use strings.SplitN(parts[0], "/", 3)` and take the last part as the "repository" no matter if there are more "nestings" with "/. " 

WDYT?

Reply:
If `Organization` is not need, then we don't need to keep track of it. This also reminds me to write some unit tests for `extractImageInfo`.

Reply:
Implemented in [5e87931](https://github.com/cloudoperators/heureka/pull/309/commits/5e8793132fa322f8c2f2ed6ab3027559f0605314)